spec/expression.dd

```diff
@@ -1615,14 +1616,14 @@ $(H3 $(LNAME2 this, this))
         }
         -------------
 
-    $(P Assignment to $(D this) is not allowed.)
+    $(P Assignment to $(D this) is not allowed for classes.)
 
 $(H3 $(LNAME2 super, super))
 
     $(P $(D super) is identical to $(D this), except that it is
         cast to $(D this)'s base class.
         It is an error if there is no base class.
-        It is an error to use $(D super) within a struct member function.
+        It is an error to use $(D super) outside a class.
```

Comment:
When you do this:
```D
class C
{
    struct S
    {
        void m() 
        {
            auto x = super;
        }
    }
}
```
The error is "`super` is only allowed in non-static class member functions". Maybe copy that wording in the spec.